pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/signature/JavaFieldSignature.java

```diff
@@ -54,10 +55,10 @@ private static int code(Visibility visibility, boolean isStatic, boolean isFinal
      *
      * @return The signature of the field
      */
-    public static FieldSignature buildFor(ASTFieldDeclaration node) {
+    public static JavaFieldSignature buildFor(ASTFieldDeclaration node) {
         int code = code(Visibility.get(node), node.isStatic(), node.isFinal());
```

Comment:
Yes, I think it's safe to keep the `code` version here. That could prevent a huge number of instance creations, and it doesn't clutter the code too much anyway

We have unit tests [here](https://github.com/pmd/pmd/blob/master/pmd-java/src/test/java/net/sourceforge/pmd/lang/java/metrics/SignatureTest.java#L209) to check that two methods or fields with the same signature share the signature instance. I don't think we could test the map specifically if that's what you're saying...